### PR TITLE
Add onload attribute to ReCaptchaTagHelper

### LINKED ISSUE #15281: Add support for the ReCaptcha onload callback
### Is your feature request related to a problem? Please describe.
I use JQuery Unobtrusive Validation to validate my forms. I also use the ReCaptcha feature to protect my forms against robots. I'd like to be able to add validation to the reCAPTCHA form element, which is loaded dynamically and thus can't be added until after the reCAPTCHA is loaded. Fortunately, Google provides an `onload` parameter that you can provide on the ReCaptcha script ([see documentation](https://developers.google.com/recaptcha/docs/display#explicit_render)). Unfortunately, there isn't a convenient way to specify that parameter and when using the `ReCaptchaTagHelper`.

### Describe the solution you'd like
Add an attribute to the `ReCaptchaTagHelper` for specifying the onload callback.

### Describe alternatives you've considered
For my site, I replaced the base Orchard Core `ReCaptchaTagHelper` with a custom implementation with the onload attribute.

### PR DESCRIPTION
**Pull request by mroskamp**
Add onload attribute to ReCaptchaTagHelper to allow specifying the onload callback to the reCAPTCHA script. fixes #15281

### CODE REVIEW

**Comment by hishamco** [comment]
Why you remove the `ILocalizationService`? the build is fail

**Comment by mroskamp** [comment]
> Why you remove the `ILocalizationService`? the build is fail

It was accidental. A fix is incoming.

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs [hunk 1]
```diff
@@ -77,12 +83,28 @@ private async Task ShowCaptcha(TagHelperOutput output)
...
-            var settingsUrl = $"{_settings.ReCaptchaScriptUri}?hl={cultureInfo.TwoLetterISOLanguageName}";
+        private async Task<string> GetReCaptchaScriptUrlAsync()
+        {
+            var query = new QueryString();
+            var cultureInfo = await GetCultureAsync();
+            if (cultureInfo != null)
+            {
+                query = query.Add("hl", cultureInfo.TwoLetterISOLanguageName);
+            }
+            if (!string.IsNullOrWhiteSpace(OnLoad))
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs:98]:
```suggestion

            if (!string.IsNullOrWhiteSpace(OnLoad))
```

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs [hunk 2]
```diff
@@ -77,12 +83,28 @@ private async Task ShowCaptcha(TagHelperOutput output)
...
+                query = query.Add("onload", OnLoad);
+            }
 
-            builder.Attributes.Add("src", settingsUrl);
-            _resourceManager.RegisterFootScript(builder);
+            var settingsUrl = new UriBuilder(_settings.ReCaptchaScriptUri)
+            {
+                Query = query.ToString()
+            };
+            return settingsUrl.ToString();
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs:107]:
```suggestion

            return settingsUrl.ToString();
```

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs [hunk 3]
```diff
@@ -77,12 +83,28 @@ private async Task ShowCaptcha(TagHelperOutput output)
             output.TagMode = TagMode.StartTagAndEndTag;
 
             var builder = new TagBuilder("script");
-            var cultureInfo = await GetCultureAsync();
+            builder.Attributes.Add("src", await GetReCaptchaScriptUrlAsync());
+            _resourceManager.RegisterFootScript(builder);
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs:87]:
```suggestion

            _resourceManager.RegisterFootScript(builder);
```

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs [hunk 4]
```diff
@@ -30,16 +30,16 @@ public class ReCaptchaTagHelper : TagHelper
...
+            ILogger<ReCaptchaTagHelper> logger,
+            ILocalizationService localizationService)
         {
             _resourceManager = resourceManager;
             _httpContextAccessor = httpContextAccessor;
             _settings = optionsAccessor.Value;
             Mode = ReCaptchaMode.PreventRobots;
-            _localizationService = localizationService;
             _logger = logger;
+            _localizationService = localizationService;
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs:42]: Revert the changes here

**Review by hishamco — CHANGES_REQUESTED** [review]

**Review by hishamco — APPROVED** [review]

**Review by sebastienros — APPROVED** [review]

**Pull request merged**